### PR TITLE
Improve request filtering process (#59)

### PR DESCRIPTION
**Pull request by johnkrovitch**

### CODE REVIEW

File: Admin/Factory/AdminFactory.php [hunk 1]
```diff
@@ -168,13 +177,19 @@ public function create($name, array $configuration)
...
+            ->create($adminConfiguration);
+
         // create Admin object
         $admin = new Admin(
             $name,
             $dataProvider,
             $adminConfiguration,
             $this->messageHandler,
-            $this->eventDispatcher
+            $this->eventDispatcher,
```
Comment (lutangar) [inline Admin/Factory/AdminFactory.php]: Could some of the admins dependencies be set earlier in the container?

File: Filter/RequestFilter.php [hunk 2]
```diff
@@ -39,6 +64,34 @@ public function filter(Request $request)
...
+    }
+
+    public function getMaxPerPage()
+    {
+        return $this->maxPerPage;
+    }
+
+    public function getCurrentPage()
+    {
+        return 1;
```
Comment (lutangar) [inline Filter/RequestFilter.php]: Hardcoded maybe?

File: Admin/Factory/AdminFactory.php (hunk 1 above)
Comment (johnkrovitch) [inline Admin/Factory/AdminFactory.php]: This is not planned for now, but I'll take a look on how I can do that

File: Filter/RequestFilter.php (hunk 2 above)
Comment (johnkrovitch) [inline Filter/RequestFilter.php]: Yess, this is a default value that does not be set in the default RequetFilter, only in the PagerFantaRequestFilter

**Comment by johnkrovitch** [comment]
@lutangar to answer to question, the RequestFilterInterface does not stand for implementing a pager, but just to implements a way to get the pager parameters from the request

**Pull request merged**